### PR TITLE
Towards #2326 - removing some namespaces

### PR DESCRIPTION
**Pull request by sfilipi**
Replacing the following namespaces as described in #2326:

Microsoft.ML.Internal.Internallearn.ResultProcessor 
Microsoft.ML.Trainers.FastTree.Internal,
Microsoft.ML.Learners, 
Microsoft.ML.Trainers.SymSGD.

### CODE REVIEW

File: src/Microsoft.ML.FastTree/Application/DominationLossApplication.cs [hunk 1]
```diff
@@ -2,7 +2,7 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-namespace Microsoft.ML.Trainers.FastTree.Internal
+namespace Microsoft.ML.Trainers.FastTree
 {
 #if OLD_DATALOAD
```
Comment (yaeldekel) [inline src/Microsoft.ML.FastTree/Application/DominationLossApplication.cs]: >OLD_DATALOAD [](start = 4, length = 12)

Is this whole file inside the #if?

File: src/Microsoft.ML.FastTree/Application/WinLossSurplusApplication.cs [hunk 2]
```diff
@@ -2,7 +2,7 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-namespace Microsoft.ML.Trainers.FastTree.Internal
+namespace Microsoft.ML.Trainers.FastTree
 {
 #if OLD_DATALOAD
```
Comment (yaeldekel) [inline src/Microsoft.ML.FastTree/Application/WinLossSurplusApplication.cs]: >OLD_DATALOAD [](start = 4, length = 12)

Seems like all the files in this folder contain only grayed out code.

File: src/Microsoft.ML.FastTree/Application/DominationLossApplication.cs (hunk 1 above)
Comment (TomFinley) [reply src/Microsoft.ML.FastTree/Application/DominationLossApplication.cs]: I think we could in principle safely remove this, but I'd make that a separate PR.

Comment (sfilipi) [reply src/Microsoft.ML.FastTree/Application/DominationLossApplication.cs]: remove the file? of the directive?

---
In reply to: [254797836](https://github.com/dotnet/machinelearning/pull/2442#discussion_r254797836) [](ancestors = 254797836)

Comment (sfilipi) [reply src/Microsoft.ML.FastTree/Application/DominationLossApplication.cs]: it looks like. 

---
In reply to: [254505167](https://github.com/dotnet/machinelearning/pull/2442#discussion_r254505167) [](ancestors = 254505167)

**Comment by sfilipi** [comment]
@yaeldekel @TomFinley when you have a chance, lmk if this looks good or anything else needs to be changed.

File: src/Microsoft.ML.Ensemble/EntryPoints/FeatureSelector.cs [hunk 3]
```diff
@@ -2,15 +2,15 @@
...
 using Microsoft.ML.EntryPoints;
 
 [assembly: EntryPointModule(typeof(AllFeatureSelectorFactory))]
 [assembly: EntryPointModule(typeof(RandomFeatureSelector))]
 
-namespace Microsoft.ML.Ensemble.EntryPoints
+namespace Microsoft.ML.Ensemble
 {
     [TlcModule.Component(Name = AllFeatureSelector.LoadName, FriendlyName = AllFeatureSelector.UserName)]
     public sealed class AllFeatureSelectorFactory : ISupportFeatureSelectorFactory
```
Comment (rogancarr) [inline src/Microsoft.ML.Ensemble/EntryPoints/FeatureSelector.cs]: Does this belong in Ensemble?

File: test/Microsoft.ML.Tests/Scenarios/Api/Estimators/IntrospectiveTraining.cs [hunk 4]
```diff
@@ -94,7 +94,7 @@ public void FastTreeClassificationIntrospectiveTraining()
         public void FastForestRegressionIntrospectiveTraining()
         {
             var ml = new MLContext(seed: 1, conc: 1);
-            var data = DatasetUtils.GenerateFloatLabelFloatFeatureVectorSamples(1000);
+            var data = SamplesUtils.DatasetUtils.GenerateFloatLabelFloatFeatureVectorSamples(1000);
```
Comment (rogancarr) [inline test/Microsoft.ML.Tests/Scenarios/Api/Estimators/IntrospectiveTraining.cs]: You're alreading `using SamplesUtils`. Maybe drop the namespace ref?

File: test/Microsoft.ML.Tests/TrainerEstimators/TreeEnsembleFeaturizerTest.cs [hunk 5]
```diff
@@ -18,7 +17,7 @@ public partial class TrainerEstimators
         public void TreeEnsembleFeaturizerOutputSchemaTest()
         {
             // Create data set
-            var data = DatasetUtils.GenerateBinaryLabelFloatFeatureVectorSamples(1000).ToList();
+            var data = SamplesUtils.DatasetUtils.GenerateBinaryLabelFloatFeatureVectorSamples(1000).ToList();
```
Comment (rogancarr) [inline test/Microsoft.ML.Tests/TrainerEstimators/TreeEnsembleFeaturizerTest.cs]: Using?

File: test/Microsoft.ML.Tests/TrainerEstimators/TreeEstimators.cs [hunk 6]
```diff
@@ -331,7 +331,7 @@ private void LightGbmHelper(bool useSoftmax, out string modelString, out List<Gb
                 floatLabels[i] = labels[i];
 
             // Allocate LightGBM data container (called Dataset in LightGBM world).
-            var gbmDataSet = new Dataset(sampleValueGroupedByColumn, sampleIndicesGroupedByColumn, _columnNumber, sampleNonZeroCntPerColumn, _rowNumber, _rowNumber, "", floatLabels);
+            var gbmDataSet = new LightGBM.Dataset(sampleValueGroupedByColumn, sampleIndicesGroupedByColumn, _columnNumber, sampleNonZeroCntPerColumn, _rowNumber, _rowNumber, "", floatLabels);
```
Comment (rogancarr) [inline test/Microsoft.ML.Tests/TrainerEstimators/TreeEstimators.cs]: Do you need the explicit namespace ref?

**Pull request merged**